stablehlo/dialect/TypeInference.h

```diff
@@ -83,19 +83,60 @@ LogicalResult verifyReducerShape(Optional<Location> loc, Block& block,
                                  ArrayRef<int64_t> allowedDimensions,
                                  bool allInputsUnranked);
 
-// Verifies replica groups attached to collective communication operations.
-// P1. 'replicaGroups' must be a 2-D tensor.
-// P2. replicaGroups' cannot be empty.
-// P3. If `allGroupsMustHaveSameSize` is true, then each group is of the same
-//     size.
-// P4. All values in `replica_groups` are unique and covers all the values in
-//     the interval [0, N-1], where N is the total number of replica ids.
-// P5. replica group size must be equal to 'expectedGroupSize'.
-LogicalResult verifyReplicaGroups(Optional<Location> location,
-                                  DenseIntElementsAttr replicaGroups,
-                                  bool allGroupsMustHaveSameSize,
-                                  bool useGlobalDeviceIds,
-                                  Optional<size_t> expectedGroupSize);
```

Comment:
Looks like we can delete pretty much all helper functions from this file, except for `inferGatherShape`?

Reply:
You are correct: every thing under "Utils for shape functions." in this header file can be deleted except for "inferGatherShape()"